crates/sui-core/src/authority/authority_per_epoch_store.rs

```diff
@@ -1480,6 +1518,15 @@ impl AuthorityPerEpochStore {
                 self.record_end_of_publish(*authority, transaction.key(), consensus_index)?;
                 Ok(None)
             }
+            ConsensusTransactionKind::CapabilityNotification(capabilities) => {
+                let authority = capabilities.authority;
```

Comment:
If I understand the logic correctly, we want to make sure that by the time we build the epoch change transaction in checkpoint builders we want all validators to agree on the same capabilities map.

If so, we need to make sure we don't process `CapabilitNotification` messages after `2f+1` EndOfPublish messages are received (otherwise you might have race conditions).

You can do this check by adding this check before processing capability notification:
```
self
                    .get_reconfig_state_read_lock_guard()
                    .should_accept_consensus_certs()
```

(You can see that handling user cert above has similar check)